### PR TITLE
[System.Data] use PAL_gssapi.c for SSPI

### PR DESCRIPTION
**Pull request by EgorBo**
PAL_gssapi.c was converted to C recently in corefx but since our fork is not up to date I copied `PAL_gssapi.c` and `PAL_gssapi.h` into mono/metadata (I guess I should just cherry-pick those files to our fork).
For System.Data it will allow users to connect to sql servers using SSPI (Security Support Provider Interface).
See https://github.com/mono/mono/issues/9028 and https://github.com/mono/mono/issues/9751
on macOS (and iOS) it uses built-in GSS.framework.
on Linux it requires an additional package (`krb`) to be installed (see .NET Core prerequisites, e.g. https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x#ubuntu). 
Unfortunately it's not installed on our CI yet.

### CODE REVIEW

**Comment by EgorBo** [comment]
@monojenkins build failed

**Comment by akoeplinger** [comment]
@monojenkins build

File: mcs/class/System.Data/Makefile [hunk 1]
```diff
@@ -25,6 +25,10 @@ ifdef MONO_FEATURE_APPLETLS
 LIB_MCS_FLAGS += -d:MONO_FEATURE_APPLETLS
 endif
 
+ifdef ENABLE_GSS
+LIB_MCS_FLAGS += -d:ENABLE_GSS
```
Comment (marek-safar) [inline mcs/class/System.Data/Makefile]: What is this for?

Comment (EgorBo) [reply mcs/class/System.Data/Makefile]: if GSS is not enabled then we do not include [DllImport] and use stubs (like it was before). As far as I remember "dead" DllImports cause problems for AOT, see https://github.com/mono/corefx/blob/master/src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.NetSecurityNative.cs#L15

**Comment by EgorBo** [comment]
@monojenkins build failed

File: mcs/class/System.Data/Makefile (hunk 1 above)
Comment (marek-safar) [reply mcs/class/System.Data/Makefile]: ENABLE_GSS won't be set for BCL builds, you need to do it manually via `mcs/build/config.make`

**Comment by marek-safar** [comment]
@monojenkins backport 2018-08

**Comment by marek-safar** [comment]
@monojenkins commit csproj

**Comment by marek-safar** [comment]
@alexischr I have merged this in. The name unification will come as part of Martin renaming work.

**Pull request merged**